### ISSUE TITLE
rundir-219.12

### ISSUE THREAD

**Issue by udeved** [andywingo/elogind#2]
I noticed 219.12 does not create a rundir on its own.

If I let it be created by eg dbus script, elogind seems to work fine. Without rundir, my logs show errors regarding missing rundir.

Leads me to another question, would it be possible to have a --with-rundir configure option?
Eg that rundir could be /run/elogind instead of /run/systemd ?

My init system is openrc.

**Comment by kensington** [comment]
I saw the same issue.

**Comment by kensington** [comment]
@udeved it looks like on guix this is created by the system, so maybe this is a distro integration problem instead: http://git.savannah.gnu.org/cgit/guix.git/tree/gnu/system/file-systems.scm#n210

**Comment by udeved** [comment]
@kensington 

Thx, I'll have a look.

I did simply put the run_dir creation in dbus openrc-run script.
Gentoo basically serves as upstream to provide alternative on arch based systems. 
Out of curiosity, did you manage to start a wayland session on gentoo with elogind?
It doesn't work for me both on gentoo and manjaro.

**Comment by kensington** [comment]
@udeved yep, although it's still a work-in-progress upstream I was able to launch KDE Plasma / Wayland session in the past with the help of elogind.

**Comment by udeved** [comment]
Perhaps the important question, did you use libinput instead of evdev for xorg input drivers?

**Comment by kensington** [comment]
libinput is installed but xf86-input-libinput is not. xorg itself wasn't given any special config one way or the other.

**Comment by udeved** [comment]
Thanks, I may try again after removing xf86-input-libinput,I thought the trouble was input drivers from the logs.

**Comment by udeved** [comment]
@kensington 

How do you approach the missing /run/systemd on gentoo?

I think its best to try run a /bin/sh check for the run_dir and if necessarey create it before elogind call in the Exec line of the dbus service?

```
Exec = /bin/bash -c "[[ ! -d /run/systemd ]] && mkdir /run/systemd";libpath/to/elogind
```

Another way would be openrc service, but I find it ugly.

**Comment by udeved** [comment]
@kensington 

This one works for me.

```
Exec=/bin/bash -c "[[ ! -d /run/systemd ]] && mkdir /run/systemd;/usr/lib/elogind/elogind"
```

Should work on gentoo with:

```
Exec=/bin/bash -c "[[ ! -d /run/systemd ]] && mkdir /run/systemd;/usr/libexec/elogind/elogind"
```

**Comment by kensington** [comment]
@udeved 

> How do you approach the missing /run/systemd on gentoo?

It's still in an experimental state (for us) so I didn't implement any fix tor that (just been creating it manually during testing).

Did you have any luck with loginctl lock-session or unlock-session working?

**Comment by udeved** [comment]
I haven't tested the unlock thingy yet.
Since we are binary, I was about to decide if ck or elogind, so this will be next to test.
I have included your pull patch already.

**Comment by udeved** [comment]
@kensington 

I have tested locking/unlocking screen in plasma5 with 219.12+dbus-conf-pk.patch.
Seems to work fine, test was on real hardware.

My configuration:

```
/etc/pam.d/system-login
-session    optional    pam_elogind.so

/etc/pam.d/sddm-greeter
session    optional    pam_elogind.so
```

I have also patched the login1 dbus service Exec line to create /run/systemd before it starts if necessary.

**Comment by udeved** [comment]
@kensington 

I am not sure I am hitting the same problem with unlock.
But if I suspend or hibernate the system, it wakes up, and auto sleeps again after 5 seconds.
As if the system rejects some credentials or so.

```
Jun 24 18:47:36 laptop kernel: elogind[2012]: Suspending system...
Jun 24 18:48:26 laptop kernel: elogind[2012]: System resumed.
Jun 24 18:48:26 laptop kernel: elogind[2012]: Suspending system...
Jun 24 18:49:15 laptop kernel: elogind[2012]: System resumed.
Jun 24 18:49:15 laptop kernel: elogind[2012]: Suspending system...
Jun 24 18:50:16 laptop kernel: elogind[2012]: System resumed.
Jun 24 18:50:16 laptop kernel: elogind[2012]: Suspending system...
```

**Comment by Yamakuzure** [comment]
The new release v226.4 at elogind/elogind creates the /run/systemd directory if it is not present, so this issue is fixed there.
